Connected System Template (CST) Examples/Google Drive (OAuth,  Sending Files and Multiple Operations)/src/main/java/com/mycorp/helloworld/templates/GoogleDriveCreateFolderIntegrationTemplate.java

```diff
@@ -0,0 +1,98 @@
+import com.appiancorp.connectedsystems.templateframework.sdk.diagnostics.IntegrationDesignerDiagnostic;
+import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
+import com.google.api.client.googleapis.json.GoogleJsonResponseException;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.drive.Drive;
+import com.google.api.services.drive.model.File;
+import com.google.common.base.Stopwatch;
+import com.mycorp.helloworld.templates.utils.CreateFolderDiagnosticsFactory;
+import com.mycorp.helloworld.templates.utils.GoogleJsonResponseExceptionHandler;
+
+@TemplateId(name="GoogleDriveCreateFolderIntegrationTemplate")
+public class GoogleDriveCreateFolderIntegrationTemplate extends SimpleIntegrationTemplate {
+
+  private static final String FOLDER_MIME_TYPE = "application/vnd.google-apps.folder";
+
+  @Override
+  protected SimpleConfiguration getConfiguration(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      PropertyPath propertyPath,
+      ExecutionContext executionContext) {
+    return integrationConfiguration.setProperties(
+        //The same key (FILE_NAME_KEY) is used in the other template to create a textProperty for document name,
+        // therefore when you toggle between operations, the value you enter in this field will persist.
+        textProperty(FILE_NAME_KEY)
+            .label("Folder Name")
+            .isRequired(true)
+            .build()
+    );
+  }
+
+  @Override
+  protected IntegrationResponse execute(
+      SimpleConfiguration integrationConfiguration,
+      SimpleConfiguration connectedSystemConfiguration,
+      ExecutionContext executionContext) {
+    String folderName = integrationConfiguration.getValue(FILE_NAME_KEY);
+    File fileMetadata = new File().setName(folderName).setMimeType(FOLDER_MIME_TYPE);
+
+    CreateFolderDiagnosticsFactory createFolderDiagnosticsFactory = new CreateFolderDiagnosticsFactory();
+
+    GoogleCredential credential = new GoogleCredential().setAccessToken(executionContext.getAccessToken().get());
+    Drive drive = new Drive.Builder(new NetHttpTransport(), JacksonFactory.getDefaultInstance(), credential).build();
+    IntegrationDesignerDiagnostic.IntegrationDesignerDiagnosticBuilder diagnosticBuilder =  IntegrationDesignerDiagnostic.builder();
+
+    Stopwatch stopwatch = Stopwatch.createStarted();
+
+    Map<String, String> diagnosticResponse = new HashMap<>();
+    Map<String, Object> result = new HashMap<>();
+
+    try {
+      File responseFile = drive
+          .files()
+          .create(fileMetadata)
+          .setFields("name, id, mimeType")
+          .execute();
+      diagnosticResponse = createFolderDiagnosticsFactory.getResponseDiagnostics(responseFile);
+      result = createFolderDiagnosticsFactory.getResult(responseFile);
```

Comment:
Feels like the diagnostic factory should just create the whole diagnostic. Sort of strange that the "DiagnosticFactory" doesn't create diagnostic objects.

Reply:
Talked in person. This would result in the factory to return an IntegrationDesignDiagnosticBuilder and this will end up having lots of boilerplate code. Therefore the decision is not to do it.